operator/internal/testenv/testenv.go

```diff
@@ -82,7 +82,7 @@ func New(t *testing.T, options Options) *Env {
 		options.Logger = logr.Discard()
 	}
 
-	host, err := k3d.GetOrCreate(options.Name)
+	host, err := k3d.GetOrCreate(options.Name, k3d.WithAgents(options.Agents))
```

Comment:
Not entirely sure why this was dropped previously, but the integration scaling test fails without bumping up the agents to have 5 nodes due to pod anti-affinity rules.